Add GoPhish tools

dav3r opened the pull request:
# <!--- Provide a general summary of your changes in the Title above -->

## 🗣 Description
This PR adds the new [cisagov/gophish-tools](https://github.com/cisagov/gophish-tools) container as a service in the Docker composition.

<!--- Describe your changes in detail -->

## 💭 Motivation and Context
The scripts within `gophish-tools` will be used by PCA operators to simplify their operations and reduce the number of manual steps in their process.

<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## 🧪 Testing
The new service was tested locally as well as in COOL `env1-staging`.  All automated tests pass, including the new and updated tests.

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## 📷 Screenshots (if appropriate)

## 🚥 Types of Changes

<!--- What types of changes does your code introduce? -->
<!--- Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

<!--- Go over all the following points, and put an `x` in all the
boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask.
We're here to help! -->

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

dav3r committed:
Add gophish-tools service that contains useful PCA/GoPhish helper scripts.

dav3r committed:
Put services in alphabetical order.

dav3r committed:
Bump version from 0.0.5 to 0.0.6

dav3r committed:
Add test function for getting the gophish-tools container.

dav3r committed:
Update test to look for 4 containers, now that gophish-tools is included.
Add new test to confirm gophish-tools container exited successfully.

dav3r committed:
Add information about gophish-tools.

dav3r committed:
Rename test to clarify what is being tested.

dav3r committed:
Reduce code owners to those who are more familiar with this repository.

jsf9k commented on docker-compose.yml:
```diff
@@ -47,6 +47,24 @@ services:
+    image: cisagov/gophish-tools:latest
+    networks:
+      front:
+        ipv4_address: 172.16.202.5
```
Why do we have to hardcode IP addresses?  Normally containers in the same network can be addressed by each other via their hostnames.

There is no guarantee that this code will never run on a machine that already has the private IP `172.16.202.3`, for example.

If you really do need to specify the IPs like this, I think the network needs [an `ipam` configuration block](https://docs.docker.com/compose/compose-file/#ipv4_address-ipv6_address).

dav3r replied:
@felddy set this up originally- I think the hardcoded IP may have been needed for the postfix container, but I'm not sure.  FWIW, the network does already have an `ipam` block - see: https://github.com/cisagov/pca-gophish-composition/blob/develop/docker-compose.yml#L89-L95

I will remove the hardcoded IPs from the non-postfix containers and make sure that everything still works.

dav3r replied:
Actually, it looks like I have to remove all of the hardcoded IPs, otherwise the postfix container IP conflicts with auto-assigned IPs.

jsf9k replied:
Ah, OK.  I somehow missed the ipam block.

I guess it's OK, but it would be a lot cleaner if it worked without fixed IPs.

dav3r replied:
Take a look at it now after https://github.com/cisagov/pca-gophish-composition/pull/12/commits/2680fd8f714390466c93c81b877c435509e1d22a and see what you think.

jsf9k replied:
I see you created commit 2680fd8, which it a big step in the right direction.  Why not just [configure the default network that Docker provides your `docker-compose` application](https://docs.docker.com/compose/networking/#configure-the-default-network) and do away with the `networks` jazz in the services altogether?

dav3r replied:
Yes, that makes lots of sense!  See https://github.com/cisagov/pca-gophish-composition/pull/12/commits/eae555546b166bdb43e6baa09754a63bd079c192.  Thanks for the suggestion!

jsf9k replied:
Thanks for implementing it.  Nice work!

jsf9k reviewed (changes requested):
Not really requesting changes, just some explanation.

dav3r committed:
Remove hardcoded IP addresses for each service- they are unnecessary.
Add a comment about postfix relaying and the docker subnet.

dav3r committed:
Modify default docker network to use desired subnet, rather than creating a separate "front" network with that subnet.
This removes the need to specify the network for each container/service. 
 Thanks to @jsf9k for this suggestion!

dav3r committed:
Update comment to reflect removal of "front" network.

felddy reviewed (approved):
🐟 🧰 🔨 🔧  == 💪

mcdonnnj reviewed (approved):
Looking good! :shipit:
